project/.github/workflows/ci.yml

```
@@ -0,0 +1,47 @@
+name: CI
+
+on: [push, pull_request]
+#on:
+#  push:
+#    branches: [master]
+#  pull_request:
+
+concurrency:
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: AvaPL/scalafmt-native-formatter@v0.1.2
+
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      SBT_NATIVE_CLIENT: true
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: coursier/cache-action@v6.3.3
+
+      - name: Resolve dependencies
+        run: sbt update
+
+      - name: Compile all sources
+        run: sbt 'all compile Test/compile'
+
+      - name: Run Scalafix
+        run: sbt 'scalafixAll --check'
+
+      - name: Run tests
+        run: sbt 'all Test/test'
```

Nit:
```suggestion
        run: sbt test
```

`all` is for running several tasks in parallel, here you run only one, and the scope is redundant